RUFAS/routines/animal/animal_health/disease.py

```diff
@@ -0,0 +1,69 @@
+from RUFAS.routines.animal.life_cycle.animal_base import AnimalBase
+
+
+class Disease:
+    """
+    Abstract class representing disease simulation.
+    """
+
+    def __init__(self) -> None:
+        # get info about specific disease set by user
+        # im = InputManager()
+        # self.duration = im.get_data("disease.specific.duration")
+        # self.outcome_distribution = im.get_data("disease.specific.outcome_distribution")
+        # self.days_to_recovery = im.get_data("disease.specific.days_to_recovery")
+        # other disease-specific data: risk factors, incidence factors, etc.
+        pass
+
+    def start_disease(self, animal: AnimalBase, simulation_day: int) -> None:
+        """Starts disease.
+
+        Parameters
+        ----------
+        animal : AnimalBase
+            The animal with the disease.
+        start_day : int
+            The date in the simulation when the disease is starting.
+        """
+        # if adding disease status to animal, update the animal class to have this attribute.
+        # animal.disease_status = 'sick'
+        # animal.disease_start_day = simulation_day
+
+    def process_disease_day(self, animal: AnimalBase) -> None:
+        """Processes animal disease for each day of the duration of the disease.
+
+        Parameters
+        ----------
+        animal : AnimalBase
+            The animal with the disease.
+        """
+        # Update daily effects here based on self.outcome_distribution (set in __init__)
+        # death, recovery, cull, remain diseased
+        # if animal is still sick:
+        # self._process_disease_effects(animal)
+        # if outcome is something other than "remain diseased":
+        # self._end_disease(animal, outcome)
+        pass
+
+    def _process_disease_effects(self, animal: AnimalBase) -> None:
+        """Modifies daily life values of animal based on disease immediate effects.
+
+        Parameters
+        ----------
+        animal : AnimalBase
+            The diseased animal.
+        """
+        # pass
+
+    def _end_disease(self, animal: AnimalBase, outcome: str) -> None:
```

Comment:
```suggestion
    def _end_disease(self, animal: AnimalBase, outcome: str) -> None:
```
Could this outcome be an enum instead of a string? I looked through the design doc, and it seems that there are a number of different outcomes that could happen for a cow, but it still might be better to define them through enums instead of strings.

Reply:
Definitely. I like this idea.